scripts/twister

```diff
@@ -1114,7 +1114,11 @@ def main():
         return
 
     if options.subset:
-        suite.instances = OrderedDict(sorted(suite.instances.items()))
+        if options.device_testing:
+            suite.instances = OrderedDict(sorted(suite.instances.items(),
```

Comment:
maybe a comment about what the sorting for the `device_testing` is trying to accomplish.  Looks like a bit of black magic to me :)

Reply:
I agree, looks the same for me;) I just brought back the code which was replaced by #27537 to be used when device testing. I added comments in the code and expanded the commit msg.